core/drivers/crypto/hisilicon/hisi_qm.c

```diff
@@ -143,49 +147,114 @@ static void qm_db(struct hisi_qm *qm, uint16_t qn, uint8_t cmd, uint16_t index,
 	io_write64(qm->io_base + QM_DOORBELL_SQ_CQ_BASE, doorbell);
 }
 
+static void qm_mb_write(struct hisi_qm *qm, struct qm_mailbox *mb)
+{
+	vaddr_t dst = qm->io_base + QM_MAILBOX_BASE;
+
+	write_64bit_pair(dst, mb->x[1], mb->x[0]);
+	dsb();
```

Comment:
I don't suppose it matters much, but wouldn't `dsb_osh()` be a better match? It's fine to keep it as it is.